src/js/node/events.js

```diff
@@ -92,9 +92,28 @@ const emitWithoutRejectionCapture = function emit(type, ...args) {
   if (events === undefined) return false;
   var handlers = events[type];
   if (handlers === undefined) return false;
-
-  for (var handler of [...handlers]) {
-    handler.apply(this, args);
+  // clone handlers array if necessary since handlers can be added/removed during the loop
+  const maybeClonedHandlers = handlers.length > 1 ? handlers.slice() : handlers;
+  for (let i = 0; i < maybeClonedHandlers.length; i++) {
+    const handler = maybeClonedHandlers[i];
+    // for performance reasons Function.call(...) is used whenever possible
+    switch (args.length) {
+      case 0:
+        handler.call(this);
+        break;
+      case 1:
+        handler.call(this, args[0]);
+        break;
+      case 2:
+        handler.call(this, args[0], args[1]);
+        break;
+      case 3:
+        handler.call(this, args[0], args[1], args[2]);
+        break;
+      default:
+        handler.apply(this, args);
+        break;
+    }
```

Comment:
☝️ Do we know what common arguments are passed to event handlers? So we can intuit possible counts to unroll to.

Reply:
I had a quick look through buns codebase and the cases with 0 or 1 arguments for `emit(...)` are by far the most common cases. Some emits did emit 2 args, but I did not find any case with 3 args yet. So from my point of view handling the cases 0-3 should probably cover 99% of all use-cases.